### ISSUE TITLE
Promoted slot

### ISSUE THREAD

**Issue by Kaiiak321** [WanderingDaniel/BookingNinjaTracker#10]
If you have a booking either side, promoted slot does not appear. You have to create the slot in open space on Gannt, amend its size and then position it into place. Even though this bit works, if a smaller party tries to click it, it sends the booking system into a spin stopping them booking altogether. If the correct siize table does book, it works fine and the promoted slot becomes a booking.

Now I know about this feature it is already HUGELY IN DEMAND and will help us fill awkward dead spaces

**Comment by WanderingDaniel** [comment]
I'm not quite sure I get what you mean with "If you have a booking either side, promoted slot does not appear." 

This is on the gantt view correct? If I drag out between two events it does offer me the promoted slot option

I'll check out min requirements now!

**Comment by Kaiiak321** [comment]
So you can only create a promoted slot by clicking and dragging if there is no table afterwards.

So if table 1 has a table booked until 7.30 and then a gap until 8.30 ... if you click and drag it won't show promote slot as an option which is its intended use. However if the 8.30 booking didn't exist you could create a one hour slot... but in this case why would you?


Obviously once the slot is created, you then have the customer facing issues

**Comment by Kaiiak321** [comment]
I think on the above I am still wrong, you can't 'butt up' to an existing booking either earlier or later... if you do... promote doesn't appear, if you have black either side it does allow you to promote

**Comment by Kaiiak321** [comment]
I'm wrong again... Have a try at clicking and dragging... there seems to be no right or wrong way of doing it. As you say, there may be certain parameters...

**Comment by WanderingDaniel** [comment]
Alright - I've refined this this so it should be more clear when it shows the promote slot. (which should be most of the time now). 

I've also set it so if the guest count is lower than the mins then it will not show the promoted slot.

**Comment by Kaiiak321** [comment]
Hi,

sorry this still isn't 100%... I created a promo slot tomorrow... 4 diners... customers can book if they are a 2... I created promo slot on a 3-4 table... dragged on gannt and selected 4 diners. the promo blob on booking says up to 4 diners but I as a 2 could book it. This slot shouldn't show unless I meet the criteria... 4 ppl.

**Comment by Kaiiak321** [comment]
I have left the slot on tomorrow table 11 at summat to ate...

**Comment by Kaiiak321** [comment]
Sorry I have moved this to 5th Jan... So I don't confuse current bookings

**Comment by WanderingDaniel** [comment]
Hey - 

So in your head the diner count should be the minimum that can book it? I think that is where we're getting a bit confused.

At the moment, will abide by the session / global min count.

**Comment by Kaiiak321** [comment]
Yes if I put 4 in the guest count that’s when it should show. If I leave it blank it should default to table limits. In this case 3/4 ppl. This as always is just my opinion. On 16 Dec 2023, at 15:03, WanderingDaniel ***@***.***> wrote:﻿
Hey -
So in your head the diner count should be the minimum that can book it? I think that is where we're getting a bit confused.
At the moment, will abide by the session / global min count.

**Comment by Kaiiak321** [comment]
Hey this is still outstanding. If I create a promoted slot for 4, on a table of 4, it will still allow a 1,2,3 to book on it. As stated it should only show if someone selects 4 guests. If my table is a 4-6 table, and i promote for 6, it should show up to 6, so a 4,5,6 could book it.
